authLdap.php

```diff
@@ -565,7 +565,7 @@ function authLdap_user_role($uid)
 	// By using this approach we are now using the order of the roles from the WP_Roles object
 	// and not from the capabilities any more.
 	$userroles = array_keys(array_intersect_key($editable_roles, $usercapabilities));
-	$role = $userroles[0];
+	$role = count($userroles) > 0 ? $userroles[0] : '';
```

Comment:
I'd write this a bit separately but in general it's a good catch! Thank you for that!
```suggestion
	$role = ($userroles !== []) ? $userroles[0] : '';
```

Reply:
Nitpick: The comaprison `$userroles !== []` is slightly slower.

````
<?php

$arr = array('a', 'b', 'c', 'd');
$tries = 1E7;

$start_time = microtime(true);
for ($i = 0; $i < $tries; $i++)
    $u = count($arr) > 0 ? $arr[0] : '';
$end_time = microtime(true);
echo $end_time - $start_time . PHP_EOL;

$start_time = microtime(true);
for ($i = 0; $i < $tries; $i++)
    $u = $arr !== [] ? $arr[0] : '';
$end_time = microtime(true);
echo $end_time - $start_time . PHP_EOL;
````

Reply:
Interesting! Thanks for pointing that out. 

Though the speed-gain is negligible. For me the better readability (The variable contain an empty array as opposed to counting the entries of the variable (Oh! that must be an array then) results in 0 which means the variable contains an empty array) is the main point to use that notation.

Reply:
Hmm. One line above, `$userroles` is assigned as the result of `array_keys`. To me, it is quite obvious that it is an array.

Reply:
Yeah. It's clear that it is an array. But the added cognitive load to figure out that `count($something) > 0` means that the array is empty over `$something === []` is - at least for me - higher. Even though it adds to the cognitive load of the computer. But that's not *my* brain :grin:

Reply:
Thanks a lot for merging. I suppose that we will use this with our next release of [our website](https://www.plus.ac.at) by the end of January.